storage-resize-images/functions/src/index.ts

```diff
@@ -49,149 +47,194 @@ logs.init();
-      return;
+  } catch (err) {
+    logs.error(err);
+  } finally {
+    if (originalFile) {
+      !verbose || logs.tempOriginalFileDeleting(filePath);
+      try {
+        fs.unlinkSync(originalFile);
+      } catch (err) {
+        logs.errorDeleting(err);
+      }
+      !verbose || logs.tempOriginalFileDeleted(filePath);
     }
-
-    if (!supportedContentTypes.includes(contentType)) {
-      logs.unsupportedType(supportedContentTypes, contentType);
-      return;
+    if (config.deleteOriginalFile === deleteImage.always) {
+      // Delete the original file
+      if (remoteFile) {
+        try {
+          logs.remoteFileDeleting(filePath);
+          await remoteFile.delete();
+          logs.remoteFileDeleted(filePath);
+        } catch (err) {
+          logs.errorDeleting(err);
+        }
+      }
     }
+  }
+};
 
-    if (
-      config.includePathList &&
-      !startsWithArray(config.includePathList, tmpFilePath)
-    ) {
-      logs.imageOutsideOfPaths(config.includePathList, tmpFilePath);
-      return;
-    }
+export const generateResizedImage = functions.storage
+  .object()
+  .onFinalize(async (object) => {
+    await generateResizedImageHandler(object);
+  });
 
-    if (
-      config.excludePathList &&
-      startsWithArray(config.excludePathList, tmpFilePath)
-    ) {
-      logs.imageInsideOfExcludedPaths(config.excludePathList, tmpFilePath);
+/**
+ *
+ */
+export const backfillResizedImages = functions.tasks
+  .taskQueue()
+  .onDispatch(async (data) => {
+    const runtime = getExtensions().runtime();
+    if (!config.doBackfill) {
+      await runtime.setProcessingState(
+        "PROCESSING_COMPLETE",
+        "Existing images were not resized."
+      );
       return;
     }
-
-    if (object.metadata && object.metadata.resizedImage === "true") {
-      logs.imageAlreadyResized();
-      return;
+    if (data == undefined) {
+      logs.startBackfill();
     }
-
-    const bucket = admin.storage().bucket(object.bucket);
-    const filePath = object.name; // File path in the bucket.
-    const parsedPath = path.parse(filePath);
-    const objectMetadata = object;
-
-    let originalFile;
-    let remoteFile;
-    try {
-      originalFile = path.join(os.tmpdir(), filePath);
-      const tempLocalDir = path.dirname(originalFile);
-
-      // Create the temp directory where the storage file will be downloaded.
-      logs.tempDirectoryCreating(tempLocalDir);
-      await mkdirp(tempLocalDir);
-      logs.tempDirectoryCreated(tempLocalDir);
-
-      // Download file from bucket.
-      remoteFile = bucket.file(filePath);
-      logs.imageDownloading(filePath);
-      await remoteFile.download({ destination: originalFile });
-      logs.imageDownloaded(filePath, originalFile);
-
-      // Get a unique list of image types
-      const imageTypes = new Set(config.imageTypes);
-
-      // Convert to a set to remove any duplicate sizes
-      const imageSizes = new Set(config.imageSizes);
```

Comment:
The extension appears to be stuck on processing 3 images. Debugging has shown:

![image](https://user-images.githubusercontent.com/2060661/203334943-149c1230-3297-4b6f-8e9b-343c3be2e9bf.png)

This will repeat on each new attempt at processing.

Reply:
The config aims to process 3 images in the query, so this could be a permanent loop of trying to process the first 3 , even when resized.

Could this be because of the initial boot up, when calling `enqueue`? 

`Deleting the original image` setting will also not work as the fn will attempt to cycle through the first 3 images, even if they are resized.

Reply:
So usually it takes a while for the function kicked off by our backend to run. That's because it does 7-10 (ish) retries lasting about 30 seconds - 1 minute, while it's waiting for the permissions we just granted (to run the function) to propagate. But if you're seeing the function actually running, then that permission has propagated. 

But the error looks like a permissions error. What is the rest of the error? Lacks ... ? I think this will help us determine if it's just waiting for a different permission to propagate or if that permission would not be granted based on the current configuration.

Reply:
First error:

![image](https://user-images.githubusercontent.com/2060661/203347900-db803e33-8013-4702-b076-517a02e98d04.png)


Seconds error:

![image](https://user-images.githubusercontent.com/2060661/203348001-c9962747-70c0-4f33-bb10-193d0f30c6e1.png)

Strangely, the fn then does try to init, but then seems to stall...

![image](https://user-images.githubusercontent.com/2060661/203349405-3cd970a0-1b0a-4b98-9f36-50767b113990.png)

Reply:
Hmmm, that Cloud tasks enqueuer permission should be granted by default by the backend... Can you take a look at the service account in Cloud console and see what roles it has?

Reply:
FWIW - i just successfully backfilled a bucket with 37 images (3 per task):
<img width="1149" alt="Screen Shot 2022-11-22 at 10 13 27 AM" src="https://user-images.githubusercontent.com/4635763/203390653-b32b7a02-316a-4fbb-8f33-9851c80fa405.png">
<img width="980" alt="Screen Shot 2022-11-22 at 10 14 37 AM" src="https://user-images.githubusercontent.com/4635763/203390689-1700db8f-16e0-4d60-b9ae-97146c8f7840.png">


This seems like a possible backend issue? Either permissions not being granted correctly, or not propagating in time?

Reply:
Can confirm his is now working after upgrading to the latest version of the CLI.

Tested on installation and configuration lifecycles.